docs/proposal/content-library-support.md

```diff
@@ -0,0 +1,257 @@
+
+#### Story 2
+
+As a cluster user, I'd like to use content library items(templates/OVAs) as the source of cluster nodes.
+
+### Overall Design
+
+The design focuses on how to clone VMs from content library items. For the content library management itself users can reference vSphere documentation.
+
+### Implementation Details/Notes/Constraints
+
+#### Naming scheme
+
+We use `Template` field of `VirtualMachineCloneSpec` to specify the source we want to clone,
+since we want to augment it to also from the content library, we need to have a way to differentiate the source.
+The proposal is to mutate the template field to accept an URI scheme, as: `scheme://location` which:
+
+* scheme, can be vsphere|library
+  * vsphere for conventional vm templates
+  * library for OVA/templates from content library
+* to make it backward compatible, the scheme part can be omitted for conventional vm templates, so if we encounter
+  template without scheme part, we assume it is vsphere template, existing CAPV does not need to do anything.
+* When a content library scheme is being used, the scheme should look like `library://[name_or_id_of_library]/item`.
+  * In case of two libraries existing with the same name, cloning a new VM should error and user should be 
+  requested to use the library UUID instead
+  * Library UUID can be queried with GoVC: `govc library.info`
+
+#### API changes
+
+##### VirtualMachineCloneSpec CR
+
+The `VirtualMachineCloneSpec` has following changes, the spec is used to specify clone spec for `VsphereVM`, it needs the following change to deal with newly added content library capabilities.
+
+```golang
+type VirtualMachineCloneSpec struct {
+  // Template is the name or inventory path of the template used to clone
+  // the virtual machine. It can be a vSphere template, or a content library item. For example:
+  // vsphere template: [vsphere://location of the vsphere template], vsphere:// can be omitted for backward compatibility
+  // content library: [library://library_name_or_id/content_library_item_name]
+  // note that this has impact on cloneMode field, since it is only possible to do full clone from content library items,
+  // and we should have validation rule for this.
+  // +kubebuilder:validation:MinLength=1
+  Template string `json:"template"`
+}
+```
+
+#### Implementation Details
+
+
+##### Getting the right Content Library item
+Because content libraries can have the same name, and items on different content libraries
+can have the same name, we should have a way to disambiguate what exact template users wants to deploy.
+
+This way, given the schema of a content library item is `library://[library_name_or_id]/itemname` we should:
+* Parse the field as a URI. If just a "Host" is present, it means user is requesting an item name, so it should be queried by item name
+* If both host and item are present, then host is the content library name (or uuid) and path is the item name
```

Comment:
Do I get it right that it would be possible to use it via (noob content library question):

* not specify the library name or id: `library://ubuntu-22.04-k8s-1.28.0`
* specify the library name or id: `library://foo/ubuntu-22.04-k8s-1.28.0`

Adding direct examples here would make it a better read :-)

Reply:
yes you do! but if there are two items named `ubuntu-22.04-k8s-1.28.0` on different libraries it should be an error, as CAPV should not select for the user one of two IMHO

Same for 2 content libraries with same names on VC.

Let me add some examples here!

Reply:
That sounds good, thx

Reply:
Sounds good, we need  a way to unique identify the library item user want, the proposed scheme library://library_name_or_id/itemname should be able to address that. library_id is what we actually needed here to resolve ambiguity here, allowing user to use library_name instead or even omitting the library_id/library_name when the item name is unique is just for enhancing user experience.
Personally I think the full path is better( although it is harder to use),  as it can avoid us problem down the road. I remember we have several similar issues for templates, as we allow user to just specify the template name not the full path of the template, and use tkr-resolver to update it to the full path later, which may break some time.

Reply:
I've discussed with @adam-jian-zhang and I'm also adding to the status field of vspherevm what was the "resolved" library id / item name

I was even thinking about adding the item ID, but I think in this case is too much, as we are not allowing users to add the item ID anyway, wdyt?

Reply:
Question: what would the status field get used for? If it is just for visualization: I think having a log line would also be okay.

Reply:
good question, IIRC vspherevm is not mutable after cloned right? So it would just be a "I'm based on this image" field, which can be a log line yes (unless I'm forgetting some other condition)

Reply:
If we don't actually use the field I slightly tend to only log it yes 👍 

I assume the image could get moved in the content library, so the status field may get outdated.